core/drivers/stm32_i2c.c

```diff
@@ -821,6 +821,9 @@ int stm32_i2c_init(struct i2c_handle_s *hi2c,
 
 	clk_disable(hi2c->clock);
 
+	if (hi2c->pinctrl && pinctrl_apply_state(hi2c->pinctrl))
+		return -1;
```

Comment:
Linked to other comment, needs to be updated to use pinctrl_apply_state() error code

Reply:
This change simply makes `stm32_i2c_init()` to report an error on pinctrl issues. I think it's simpler, as a fixup commit for coming next release tag. Changing all stm32_i2c API function to use `TEE_Result` code instead of `int` value is a good idea but i'll rather treat that after next tag and on a specific P-R, if you're ok.

Reply:
Ok :)